module/spl/spl-taskq.c

```diff
@@ -763,11 +763,12 @@ taskq_thread_spawn_task(void *arg)
 	taskq_t *tq = (taskq_t *)arg;
 	unsigned long flags;
 
-	(void) taskq_thread_create(tq);
-
-	spin_lock_irqsave_nested(&tq->tq_lock, flags, tq->tq_lock_class);
-	tq->tq_nspawn--;
-	spin_unlock_irqrestore(&tq->tq_lock, flags);
+	if (taskq_thread_create(tq) == NULL) {
+		/* restore spawning count if failed */
```

Comment:
I wondered at first the context of this comment (when looking at the original patch) and thought "restore to _what_"?  How about:

    /* cancel the request to spawn if thread creation fails */